code/game/gamemodes/wizard/spellbook.dm

```diff
@@ -492,6 +517,8 @@
 			E = entries[text2num(href_list["buy"])]
 			if(E && E.CanBuy(H,src))
 				if(E.Buy(H,src))
+					if(E.limit)
+						E.limit--
 					uses -= E.cost
 		else if(href_list["refund"])
 			E = entries[text2num(href_list["refund"])]

```

Comment:
I'm not sure if you did this intentionally but I'd like to point out that you didn't port the lines increasing the limit after refund (https://github.com/tgstation/-tg-station/pull/10675/files#diff-3cf4f9f2465db526d14353371941f88eR565)